src/data/db.ts

```diff
@@ -0,0 +1,202 @@
+import { Promise, Thenable } from "bluebird";
+import * as pgPromise from "pg-promise";
+import * as R from "ramda";
+
+import { database } from "../config";
+import { slugify } from "../app/helpers";
+
+// Limit the amount of debugging of SQL expressions
+const trimLogsSize: number = 200;
+const maxRecordsPerPage: number = 30;
+
+const pgpDefaultConfig = {
+  promiseLib: Promise,
+  query: (query: { query: string }): void => {
+    // Log all querys
+    console.log("[SQL   ]", R.take(trimLogsSize, query.query));
+  },
+  error: (err: Error, e: { query?: string }): void => {
+    // On error, please show me the SQL
+    if (e.query) {
+      console.error("[SQL   ]", R.take(trimLogsSize, e.query), err);
+    }
+  },
+};
+
+console.info(
+  "Connecting to the database:",
+  `${database.user}@${database.host}:${database.port}/${database.database}`
+);
+
+const pgp = pgPromise(pgpDefaultConfig);
+const db = pgp(database);
+
+interface GithubUsers {
+  id: number;
+  name: string;
+  login: string;
+  company: string;
+  blog: string;
+  email: string;
+  location: string;
+  bio: string;
+  hireable: boolean;
+  languages?: string[];
+}
+
+function findLanguagesBySlug(slugs: string[]) {
+  return db.manyOrNone(
+    "SELECT id, slug FROM programming_language WHERE slug IN ($1:csv);",
+    slugs
+  );
```

Comment:
This is just a nitpick, but we prefer passing an object to the query and parametrize the query with its properties, instead of using order based indexing.
You can read more about it here: https://github.com/vitaly-t/pg-promise#named-parameters

Reply:
I see. The slugs variable is just an array of strings, to use named parameters (from what I could gather) I would need to map it into an array of objects first. Would you even them prefer to use it in this way?